File: Problems 0400-0499/452. Minimum Number of Arrows to Burst Balloons/solution.cpp
class Solution {
    public:
        int findMinArrowShots(vector<vector<int>>& points) {
            int n = points.size();
            if (n == 0) return 0;
    
            sort(points.begin(), points.end(), [] (const vector<int>& a, const vector<int>& b) {
                return a[1] < b[1];
            });
    
            int arrow = points[0][1];
            int res = 1;
    
            for (int i = 1; i < n; i++) {
                if (points[i][0] > arrow) {
                    res++;
                    arrow = points[i][1];
                }
            }
            return res;
        }
    };

File: Problems 0400-0499/452. Minimum Number of Arrows to Burst Balloons/solution.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        n = len(points)
        if n == 0:
            return 0
        # Sort by end so we can hit at the end to maximize
        # number of balloons overlapping
        points.sort(key=lambda b: b[1])

        arrow = points[0][1]
        res = 1
        for i in range(1, n):
            start, end = points[i]
            if start > arrow:
                # have to shoot new arrow
                arrow = end
                res += 1
        return res